foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiFunctionCall.java

```diff
@@ -22,4 +26,35 @@ public class OpenAiFunctionCall implements OpenAiToolCall {
 
   /** The arguments for the function call, encoded as a JSON string. */
   @Nonnull String arguments;
+
+  /**
+   * Parses the arguments, encoded as a JSON string, into a {@code Map<String, Object>}.
+   *
+   * @return a map of the arguments
+   * @throws IllegalArgumentException if parsing fails
+   * @since 1.7.0
+   */
+  @Nonnull
+  public Map<String, Object> getArgumentsAsMap() throws IllegalArgumentException {
```

Comment:
```suggestion
  Map<String, Object> getArgumentsAsMap() throws IllegalArgumentException {
```

Reply:
I decided to remove the non-productive code.